dpg(board review): ConfidentialLedger

archerzz opened the pull request:
## Description

- update to latest swagger api which contains `GetCollections` operation
- add convenience methods of `GetReceipt` operation

related to https://github.com/Azure/autorest.csharp/issues/2199

archerzz committed:
update to latest swagger api

we need to update to the latest swagger api which contains `GetCollections` operation

archerzz committed:
add convenience method of `GetReceipt`

archerzz committed:
update api

archerzz committed:
update test recordings

archerzz committed:
add convenience methods of `GetConstituion` and `GetCollections`

archerzz committed:
add convenience method of `GetLedgerEntries`

azure-sdk commented:
**API change check**

APIView has identified API level changes in this PR and created following API reviews.

[Azure.Security.ConfidentialLedger](https://apiview.dev/Assemblies/Review/6c11d7337755493bbe6942126b633be1)

archerzz committed:
add work-around to simplify the apiview for arch board review

this is just a temporary change to roll back the parameter name `collectionId` to `subLedgerId` which was in old swagger spec

archerzz committed:
fix incorrect `Value` sufix for plural forms

archerzz commented:
close since board review is completed
